signing/src/main/java/tech/pegasys/web3signer/signing/config/DefaultArtifactSignerProvider.java

```diff
@@ -25,35 +25,55 @@
 import java.io.File;
 import java.nio.file.Path;
 import java.util.Collection;
-import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 import java.util.function.BiFunction;
+import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+/**
+ * Default implementation of {@link ArtifactSignerProvider} that loads signers and proxy signers.
+ * This class is designed to provide concurrent access to signers and proxy signers, ensuring thread
+ * safety and efficient read operations. The {@code load()} method is called infrequently, typically
+ * at startup or via a reload API call, while the getter methods are called frequently.
+ *
+ * <p>The class uses a {@link ConcurrentHashMap} for storing signers and proxy signers to allow
+ * efficient concurrent read access. It also uses a single-threaded executor to ensure that the
+ * {@code load()} method and other write operations are executed sequentially.
+ */
 public class DefaultArtifactSignerProvider implements ArtifactSignerProvider {
 
   private static final Logger LOG = LogManager.getLogger();
+
+  private final boolean reloadKeepStaleKeys;
   private final Supplier<Collection<ArtifactSigner>> artifactSignerCollectionSupplier;
-  private final Map<String, ArtifactSigner> signers = new HashMap<>();
-  private final Map<String, Set<ArtifactSigner>> proxySigners = new HashMap<>();
-  private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+  private final Optional<Consumer<Set<String>>> postLoadingCallback;
   private final Optional<KeystoresParameters> commitBoostKeystoresParameters;
 
+  private final ConcurrentMap<String, ArtifactSigner> signers = new ConcurrentHashMap<>();
```

Comment:
What impact does switching to ConcurrentMap have on the startup time for Web3Signer when we have a lot of keys?

Reply:
I will create a performance test (outside of this PR) loading of about 5000 fully encrypted keystores load time at start up.